mlflow/server/js/src/model-registry/components/SchemaTable.tsx

```diff
@@ -23,20 +26,93 @@ type Props = {
   };
 };
 
+// return a formatted string representation of the column type
+function getColumnTypeRepr(columnType: ColumnType, indentationLevel: number): string {
+  const { type } = columnType;
+
+  const indentation = ' '.repeat(indentationLevel * INDENTATION_SPACES);
+  if (type === 'object') {
+    const propertyReprs = Object.keys(columnType.properties).map((propertyName) => {
+      const property = columnType.properties[propertyName];
+      const requiredRepr = property.required ? '' : ' (optional)';
+      const propertyRepr = getColumnTypeRepr(property, indentationLevel + 1);
+      const indentOffset = (indentationLevel + 1) * INDENTATION_SPACES;
+
+      return `${' '.repeat(indentOffset)}${propertyName}: ${
+        propertyRepr.slice(indentOffset) + requiredRepr
+      }`;
+    });
+
+    return `${indentation}{\n${propertyReprs.join(',\n')}\n${indentation}}`;
+  }
+
+  if (type === 'array') {
+    const indentOffset = indentationLevel * INDENTATION_SPACES;
+    const itemsTypeRepr = getColumnTypeRepr(columnType.items, indentationLevel).slice(indentOffset);
+    return `${indentation}Array(${itemsTypeRepr})`;
+  }
+
+  return `${indentation}${type}`;
+}
+
+function formatColumnName(spec: ColumnSpec | TensorSpec): React.ReactElement {
+  let required = true;
+  if (spec.required !== undefined) {
+    ({ required } = spec);
+  } else if (spec.optional !== undefined && spec.optional) {
+    required = false;
+  }
+  const requiredTag = required ? (
+    <Text bold>{'(required)'}</Text>
+  ) : (
+    <Text color='secondary'>{'(optional)'}</Text>
+  );
+
+  const name = 'name' in spec ? spec.name : '-';
+
+  return (
+    <Text>
+      {name} {requiredTag}
+    </Text>
+  );
+}
+
+function formatColumnSchema(spec: ColumnSpec | TensorSpec): React.ReactElement {
+  let repr = '';
+  if (spec.type === 'tensor') {
+    repr = `Tensor (dtype: ${spec['tensor-spec'].dtype}, shape: [${spec['tensor-spec'].shape}])`;
+  } else {
+    repr = getColumnTypeRepr(spec, 0);
+  }
```

Comment:
```suggestion
  const repr = spec.type === 'tensor' ? getColumnTypeRepr(spec, 0) : getTensorTypeRepr(spec)
```

I'd define `getTensorTypeRepr` and use a ternary operator to make the code a little cleaner.